### ISSUE TITLE
[v2.5.1] Launch `kubectl` Issues since update to Rancher 2.5.1

### ISSUE THREAD

**Issue by asemen2608** [rancher/rancher#29511]
**What kind of request is this (question/bug/enhancement/feature request):**

bug

**Steps to reproduce (least amount of steps as possible):**
12745
Run admin and non-admin User in Rancher 2.5.1 UI

**Result:**

- Admin:
The window admin users see when attempting to launch the kubectl window this is displayed for about 40 seconds until eventually, the command prompt appears to see  
`admin.png` screenshot. 
See the attached images:
 `user.png` 

- Non-admin: 
 Non-admin users see when attempting to launch a kubectl window from within rancher.
 Utilizing the new `Cluster Explorer` to launch a kubectl window creates a pop-up that immediately closes. 
 `user.png` references the message 

Note that kubectl is working from a user's command line (i.e not using rancher GUI)

**Other details that may be helpful:**

**Environment information**
- Rancher version 2.5.1
  after upgrade

Versions:
![version](https://user-images.githubusercontent.com/61939253/95994237-57d92600-0e30-11eb-916a-adcef397a9c9.png)

`admin.png` screenshot. 
![admin](https://user-images.githubusercontent.com/61939253/95994242-5871bc80-0e30-11eb-9479-eb500b98d542.png)

non-admin screenshot  `user.png` 
![user](https://user-images.githubusercontent.com/61939253/95994240-57d92600-0e30-11eb-9550-5d650d621575.png)

gzrancher/rancher#12745

**Comment by deniseschannon** [comment]
Can we get more details on the setup?

Management Server info? What kind of k8s cluster is it running on? Or how is it installed? Docker run? HA? 
Downstream clusters info? What kind of k8s cluster?

Which clusters are they showing the slow connection for admin and non connection for standard users? Is it the local cluster or the downstream clusters? Who are the users and how are they given access to the k8s cluster?

**Comment by bmdepesa** [comment]
I was not able to reproduce this in an upgrade from v2.4.8 -> v2.5.1

* Deploy Rancher v2.4.8 in an HA install
* Create a standard user
* Create a custom cluster as the standard user
* Upgrade to v2.5.1
* As global admin
  * local cluster kubectl shell functions in Ember UI: `kubectl get nodes` returns output
  * downstream cluster kubectl shell functions in Ember UI
  * Dashboard shell opens and functions in both local and downstream clusters
  * Add the standard user as a cluster owner to the local cluster
* Login as the standard user
  * local cluster kubectl shell functions in Ember UI
  * downstream cluster kubectl shell functions in Ember UI
  * Dashboard shell opens and functions in both local and downstream clusters

**Comment by asemen2608** [comment]
Management Server:
What kind of k8s cluster is it running on and how is it installed?
3 node RKE cluster, installed following the instructions on the rancher site

Is Docker running?
Docker is running on all nodes

Are you running a Rancher HA set up?
Yes

Downstream clusters info:
What kind of k8s cluster?
Rancher standard cluster, installed on ubuntu 18.04 VMs running docker

Which clusters are you showing/seeing the slow connection for admin and non connection for standard users?
This would be across all three non-prod clusters (DEV, CI and UAT) connecting to this rancher instance. 
Irrespective of current kubernetes versions (when issue first came to light, DEV was on v19 while CI and UAT was on v1.18, all three are now running v1.19)

Is it the local cluster or the downstream clusters?
Downstream cluster, non-admin users do not have access to the local cluster

Who are the users and how are they given access to the k8s cluster?
Added via AD/LDAP with `user-base` as their primary role. 
They are then added to specific projects in each cluster with a set of powers.

Regarding this test:
```
We were not able to reproduce this in an upgrade from v2.4.8 -> v2.5.1
The steps we did were:
Deploy Rancher v2.4.8 in an HA install
Create a standard user
Create a custom cluster as the standard user
Upgrade to v2.5.1
As global admin
local cluster kubectl shell functions in Ember UI: kubectl get nodes returns output
downstream cluster kubectl shell functions in Ember UI
Dashboard shell opens and functions in both local and downstream clusters
Add the standard user as a cluster owner to the local cluster
Login as the standard user
local cluster kubectl shell functions in Ember UI
downstream cluster kubectl shell functions in Ember UI
Dashboard shell opens and functions in both local and downstream clusters
```
Add the standard user as a cluster owner to the local cluster -> this line would not be representative of our setup (as  non-admin users are never cluster owners and would not have the powers this adds)

**Comment by bmdepesa** [comment]
@asemen2608 
```
Added via AD/LDAP with user-base as their primary role.
They are then added to specific projects in each cluster with a set of powers.
```
Can you provide the permissions list for one of the users where you see the issue? I will try to reproduce with this permission set.

**Comment by asemen2608** [comment]
Unfortunately, the rancher UI just shows `Custom` and not the specific custom roles added (custom_roles.png).

See added a screenshot for the most likely permissions they have in this specific cluster (likely_custom_roles). 
Is there a way to get specific user roles via the API?

![likely_custom_roles](https://user-images.githubusercontent.com/61939253/96618730-96c21c80-1305-11eb-9b37-8dfd5d8fcc4c.png)
![custom_roles](https://user-images.githubusercontent.com/61939253/96618733-975ab300-1305-11eb-90a6-969a055e241b.png)

**Comment by asemen2608** [comment]
In addition they have these cluster-specific roles (across all Projects)

![cluster_roles](https://user-images.githubusercontent.com/61939253/96618833-bce7bc80-1305-11eb-9663-37540ebd6075.png)

**Comment by bmdepesa** [comment]
I am able to reproduce this with a simple set of permissions. 

* Deploy Rancher in HA - v2.4.8
* Create a new user with `user-base` 
* Create a cluster
* Add the user as a cluster member with custom roles: `View Nodes, View All Projects`
![Screen Shot 2020-10-21 at 11 50 15 AM](https://user-images.githubusercontent.com/21316915/96769643-4219b280-1394-11eb-815e-c47b9b6fac2b.png)
* Login as the created user
* Open the embedded kubectl shell and see it connect - `kubectl get nodes` returns output
* Upgrade Rancher to v2.5.1
* As the created user, attempt to open the embedded kubectl
* Connection cannot be established
![Screen Shot 2020-10-21 at 11 49 55 AM](https://user-images.githubusercontent.com/21316915/96769804-78efc880-1394-11eb-8eec-df892390f740.png)

Error establishing the websocket
```
Request URL: wss://url/v3/clusters/c-sfx6p?shell=true
Error during WebSocket handshake: Unexpected response code: 404
```

The same occurs for the shell from Dashboard.

**Comment by izaac** [comment]
Reproduced the issue in `v2.5.2-rc3` a 500 response was returned from the API and `Close Code 1006` was shown in the UI.

Validated the fix in the following versions:

Rancher version: `v2.5-head (10/27/2020)` _7eb1792_
Rancher version: `master-head (10/27/2020)` _3d755ff_

* Deploy Rancher in HA k8s 
* Create a new user with `user-base` role
* Create a cluster
* Add the user as a cluster member with custom roles: `View Nodes, View All Projects`
* Login as the created user
* Open the embedded kubectl shell and see it connect - `kubectl get nodes` returns output

**Comment by asemen2608** [comment]
Will the fix be released in Rancher v2.5.3 ?
